webminer/shared/use_case.py

```diff
@@ -1,15 +1,37 @@
+"""Create a class for applying use cases
+
+Raises:
+    NotImplementedError: If a use case cannot be implemented
+
+Returns:
+    class: UseCase class
+"""
+
 from webminer.shared import response_object as res
 
 
 class UseCase(object):
+    """Creates the UseCase class
+
+    Args:
+        object (obj): A base object
+
+    Raises:
+        NotImplementedError:
+            If a a requested object cannot be implemented by the UseCase class
+
+    Returns:
+        class: Methods for executing and processing incoming requests
+    """
+
     def execute(self, request_object):
         if not request_object:
             return res.ResponseFailure.build_from_invalid_request_object(request_object)
         try:
             return self.process_request(request_object)
         except Exception as exc:
```

Comment:
@liadmagen 
```python
except Exception as exc:
```
This rises the linting issue: 
`W0703:Catching too general exception Exception`

What would you suggest?
- [ ] Fix W0703:Catching too general exception Exception

Reply:
Good point. On one hand, it's bad practice to generalize the exception so much. But on the other hand, this is pretty high in the chain - it's process the request. There are plenty of things that can go wrong.

Maybe it worth to analyze the specific parts that could go wrong and address them separately there.

Reply:
so should I rewrite tests? 
I think the easiest and fastest way would be to just create a linting exception for this error. 
You decide @liadmagen  :)